lib/ransack/adapters/active_record/ransack/nodes/condition.rb

```diff
@@ -8,6 +8,7 @@ def arel_predicate
           if negative? && attribute.associated_collection?
             query = context.build_correlated_subquery(association)
             context.remove_default_scope(association)
+            association.reflection.options[:with_deleted] = 1
```

Comment:
So, this fixes the last remaining problem with acts_as_paranoid. Sort of... It seems this change on the association gets cached. How is the association getting cached? When I run the tests from guestfolio, the deleted_at test passes if I run it in isolation. However it fails if I run the whole spec. 
Also, obviously this is a total hack! But so is this entire thing! However, this one-liner is a simple workaround - it's just where's the right place to put it?

Reply:
Oh, neat. I guess there's no good way to get from our default scope conditions to this options key. 

At least this line could probably move into the `context.remove_default_scope` method if we have nothing better. Does `1` actually mean `true` here?

One thought: could we register a hook method on our model similar to [other options][1] such as:

```
class Post < ActiveRecord::Base
  def self.ransack_without_default_scope
    with_deleted
  end
end
```

The default implementation would just return self unchanged, but this would allow setting it to remove any actual default scope. Then, we might not even need to implement `remove_default_scope` in ransack, leaving it to the end-user to declare how to negate it based on what gem they might be using (ActsAsParanoid, Paranoia, etc...)

[1]: https://github.com/activerecord-hackery/ransack#authorization-whitelistingblacklisting

Reply:
We still want to remove default scope in ransack to have a proper search query - so I believe anyway. The problem with acts_as_paranoid is not really about the default scope but with how it injects it's own logic: 
        constraint = constraint.and(table[reflection.klass.paranoid_column].eq(nil))

But I'll have a look

Reply:
Yes I'm thinking this would still be within Ransack, just saying we could require declaring the "default scope removal" method on the ransackable model, since Ransack can't know what method or gems are being used. Either as a symbol (`:with_deleted`), the evaluated condition to be removed (`table['deleted_at'].eq(nil)`), or both as a key/val: `:with_deleted => table['deleted_at'].eq(nil)`

Reply:
Scratch this.

Just discussed with @PMK9 , it won't help because `acts_as_paranoid` overrides the build_constraint method where it gets added to the join constraint (`INNER JOIN <table_name> ON <constraints>`), so just passing the scope removal to ransack is not enough: that would just remove it from the `WHERE`, not the `JOIN`.

Setting `with_deleted: true` on the reflection to remove the default scope from the join constraint would inject that setting into the class macro, e.g. `has_many :comments, with_deleted: true` so it suddenly becomes global.